modules/services/pizauth.nix

```diff
@@ -109,56 +110,59 @@ in
 
     home.packages = [ cfg.package ];
 
-    xdg.configFile."pizauth.conf".source =
-      let
-        indent = "  ";
-
-        renderScopes =
-          scopes:
-          let
-            quoted = map (s: "\"${s}\"") scopes;
-            joined = lib.concatStringsSep ",\n${indent}${indent}" quoted;
-          in
-          "[\n${indent}${indent}${joined}\n${indent}]";
-
-        renderAccount =
-          name: acc:
-          ''
-            account "${name}" {
-            ${indent}auth_uri = "${acc.authUri}";
-            ${indent}token_uri = "${acc.tokenUri}";
-            ${indent}client_id = "${acc.clientId}";
-            ${indent}client_secret = "${acc.clientSecret}";
-          ''
-          + lib.optionalString (acc.scopes != [ ] && acc.scopes != null) ''
-            ${indent}scopes = ${renderScopes acc.scopes};
-          ''
-          + lib.optionalString (acc.loginHint != "" && acc.loginHint != null) ''
-            ${indent}login_hint = "${acc.loginHint}";
-          ''
-          + lib.optionalString (acc.extraConfig != "" && acc.extraConfig != null) (
+    xdg.configFile."pizauth.conf" = {
+      onChange = "${lib.getExe cfg.package} reload";
+      source =
+        let
+          indent = "  ";
+
+          renderScopes =
+            scopes:
             let
-              indentedExtraConfig = lib.concatMapStringsSep "\n" (
-                line: if line == "" then "" else "${indent}${line}"
-              ) (lib.splitString "\n" acc.extraConfig);
+              quoted = map (s: "\"${s}\"") scopes;
+              joined = lib.concatStringsSep ",\n${indent}${indent}" quoted;
             in
-            indentedExtraConfig
-          )
-          + ''
-            }
-          '';
-
-        renderedAccounts = lib.concatStringsSep "\n" (
-          lib.mapAttrsToList (name: acc: renderAccount name acc) cfg.accounts
-        );
-
-      in
-      pkgs.writeTextFile {
-        name = "pizauth.conf";
-        text =
-          lib.optionalString (cfg.extraConfig != null && cfg.extraConfig != "") "${cfg.extraConfig}\n"
-          + renderedAccounts;
-      };
+            "[\n${indent}${indent}${joined}\n${indent}]";
+
+          renderAccount =
+            name: acc:
+            ''
+              account "${name}" {
+              ${indent}auth_uri = "${acc.authUri}";
+              ${indent}token_uri = "${acc.tokenUri}";
+              ${indent}client_id = "${acc.clientId}";
+              ${indent}client_secret = "${acc.clientSecret}";
```

Comment:
In case `clientSecret` is null, I think this should then not be rendered

Reply:
Fixed! (Had some changes uncommited ;)